tests/generic/vagga.yaml

```diff
@@ -338,42 +322,52 @@ commands:
       echo Verbosity: "$VAGGAOPT_VERBOSE"
 
   tagged: !Supervise
+    mode: wait-all-successful
     children:
       first: !Command
         container: busybox
         tags: [first_and_second, first_and_third]
         run: |
           echo hello
-          sleep 0.2
       second: !Command
         container: busybox
         tags: [first_and_second]
         run: |
-          sleep 0.05
           echo world
-          sleep 0.15
       third: !Command
         container: busybox
         tags: [first_and_third, third_only]
         run: |
-          sleep 0.1
           echo ":)"
-          sleep 0.1
 
   one-kills-another: !Supervise
+    options: |
+      Usage: vagga one-kills-another [options]
+
+      Options:
+        --exit-code <code>  First process exit code
     children:
       dying: !Command
         container: busybox
+        volumes:
+          /srv: !Persistent fail-fast
         run: |
           echo hello
-          sleep 0.1
           echo world
+          while [ ! -f /srv/hanging.child ]; do
+            sleep 0.1
+          done
+          rm -f /srv/hanging.child
```

Comment:
This will work on github actions, but may fail (i.e produce unexpected success) locally and on our buildbot CI.

Can we use some flock command, which is more reliable?

Reply:
Why may it fail? We always remove `.vagga` directory before running tests.

Cannot thing of how `flock` cannot be used here. We need something like java's `CountDownLatch`.

Reply:
Ah, yes, sorry. Looks good.